tests/src/gui/testqgsgdalguiutils.cpp

```diff
@@ -0,0 +1,335 @@
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstest.h"
+#include "qgsgdalguiutils.h"
+#include "qgsproject.h"
+#include "qgsauthmethod.h"
+#include "qgsauthconfig.h"
+#include "qgsapplication.h"
+#include "qgsauthmanager.h"
+
+class TestQgsGdalGuiUtils : public QObject
+{
+    Q_OBJECT
+
+  public:
+    TestQgsGdalGuiUtils();
+
+  private slots:
+    void initTestCase();
+    void cleanupTestCase();
+
+    void checkUriBuilding();
+
+  private:
+    void cleanupTempDir();
+
+    QList<QgsAuthMethodConfig> registerAuthConfigs();
+
+  private:
+    QString mTempDir;
+    const char *mPass = nullptr;
+};
+
+TestQgsGdalGuiUtils::TestQgsGdalGuiUtils()
+  : QObject()
+  , mTempDir( QDir::tempPath() + "/auth" )
+  , mPass( "pass" )
+{
+}
+
+void TestQgsGdalGuiUtils::initTestCase()
+{
+  cleanupTempDir();
+
+  // make QGIS_AUTH_DB_DIR_PATH temp dir for qgis-auth.db and master password file
+  const QDir tmpDir = QDir::temp();
+  QVERIFY2( tmpDir.mkpath( mTempDir ), "Couldn't make temp directory" );
+  qputenv( "QGIS_AUTH_DB_DIR_PATH", mTempDir.toLatin1() );
+
+  // init app and auth manager
+  QgsApplication::init();
+  QgsApplication::initQgis();
+  QVERIFY2( !QgsApplication::authManager()->isDisabled(), "Authentication system is DISABLED" );
+
+  // verify QGIS_AUTH_DB_DIR_PATH (temp auth db path) worked
+  Q_NOWARN_DEPRECATED_PUSH
+  const QString db1( QFileInfo( QgsApplication::authManager()->authenticationDatabasePath() ).canonicalFilePath() );
+  Q_NOWARN_DEPRECATED_POP
+  const QString db2( QFileInfo( mTempDir + "/qgis-auth.db" ).canonicalFilePath() );
+  QCOMPARE( db1, db2 );
+
+  // verify master pass can be set manually
+  // (this also creates a fresh password hash in the new temp database)
+  QVERIFY2( QgsApplication::authManager()->setMasterPassword( mPass, true ), "Master password could not be set" );
+  QVERIFY2( QgsApplication::authManager()->masterPasswordIsSet(), "Auth master password not set from passed string" );
+
+  // create QGIS_AUTH_PASSWORD_FILE file
+  const QString passfilepath = mTempDir + "/passfile";
+  QFile passfile( passfilepath );
+  if ( passfile.open( QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate ) )
+  {
+    QTextStream fout( &passfile );
+    fout << QString( mPass ) << "\r\n";
+    passfile.close();
+    qputenv( "QGIS_AUTH_PASSWORD_FILE", passfilepath.toLatin1() );
+  }
+  // qDebug( "QGIS_AUTH_PASSWORD_FILE=%s", qgetenv( "QGIS_AUTH_PASSWORD_FILE" ).constData() );
+
+  // re-init app and auth manager
+  QgsApplication::quit();
+  // QTest::qSleep( 3000 );
```

Comment:
None of this is required, there's no tests which are actually taking advantage of the auth db

```suggestion
```

Reply:
> None of this is required, there's no tests which are actually taking advantage of the auth db

If we set flag _expandAuthConfig_ and set some config-string to _configId_ we enter this block:
```

// Update URI with authentication information
if ( !configId.isEmpty() )
{
  if ( expandAuthConfig )
  { 
    QStringList connectionItems;
    connectionItems << uri;
    if ( QgsApplication::authManager()->updateDataSourceUriItems( connectionItems, configId, QStringLiteral( "ogr" ) ) )
    {
      uri = connectionItems.join( QString() );
    }
  }
  else
  {
    uri += QStringLiteral( " authcfg='%1'" ).arg( configId );
  }
}
``

So there is possibility or combination of circumstances in which we need authentication manager? 

If so - these tests are prerequisites/preparations to initializing and using it, if i understand correctly, since it is not self-contained (i.e. couchDB test was failing until authentication manager was initialized correctly with authentication method and master setting of masterpassword). 

If not - i will remove them according your suggestions, just making sure that i understand correctly

Reply:
I'm referring to the fact that none of the tests introduced here actually require the auth configuration. Theres '// Test with configId and expandAuthConfig = false', but that isn't requiring the auth. 

Then there's `QgsGdalGuiUtils::createProtocolURI( QLatin1String( "CouchDB" ), QLatin1String( "mydb" ), QLatin1String( "ab34567" ), QString(), QString(), true );`, but that's not actually doing any replacement either.